### PR TITLE
Support chroot'ing into a snapshot

### PR DESCRIPTION
**Pull request by zdykstra**

### CODE REVIEW

File: 90zfsbootmenu/zfsbootmenu-lib.sh [hunk 1]
```diff
@@ -62,25 +62,35 @@ mount_zfs() {
   mnt="${BASE}/${fs}/mnt"
   test -d "${mnt}" || mkdir -p "${mnt}"
 
+  # @ always denotes a snapshot
+  if [[ "${fs}" =~ @ ]]; then
+    is_snapshot=1
```
Comment (ahesford) [inline 90zfsbootmenu/zfsbootmenu-lib.sh:67]: Add this to the `local` list

**Review by ahesford — APPROVED** [review]
Nice work, this could be a useful tool for diagnostics.

**Comment by zdykstra** [comment]
Moved the `/tmp` bind mount to `/tmp` inthe chroot, instead of `/var/tmp`. There's no functional reason to have `/var/tmp` read-write. The text displayed when entering the chroot now mentions that `/tmp` is shared and read/write.

File: 90zfsbootmenu/zfs-chroot.sh [hunk 2]
```diff
@@ -17,19 +17,17 @@ if mountpoint="$( allow_rw=yes mount_zfs "${selected}" )"; then
   mount -t proc proc "${mountpoint}/proc"
   mount -t sysfs sys "${mountpoint}/sys"
   mount -B /dev "${mountpoint}/dev"
-  mount -B /tmp "${mountpoint}/var/tmp"
+  mount -B /tmp "${mountpoint}/tmp"
```
Comment (ahesford) [inline 90zfsbootmenu/zfs-chroot.sh:20]: I never noticed that this was already bound on `/var/tmp`, but I think `/tmp` is more intuitive.

**Pull request merged**